internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -1102,6 +1138,30 @@ func generateKongRouteModifierFromExtensionRef(pluginNamesFromExtensionRef []str
 	}
 }
 
+// hostPortFromHTTPPathModifier returns the part of scheme and hostport in the URL of `Location` header
+// for generating configuration of Kong plugins to satisfy `requestRedirect` filter of HTTPRoute.
+func schemeHostPortFromHTTPPathModifier(modifier *gatewayapi.HTTPRequestRedirectFilter) (string, string) {
+	// Since the `redirect` plugin requires the scheme, we do not change the logic to use `http` scheme when scheme is empty.
+	// TODO: According to the spec of gateway API, the original scheme should be kept if no scheme is given.
+	// So the logic may be changed if Kong gateway supports.
```

Comment:
What is left in this TODO? To provide support for leaving the scheme unset if Kong supports it? If so then we should create a tracking issue and link it here (and a related FTI requesting it from Gateway).

Reply:
Added https://github.com/Kong/kubernetes-ingress-controller/issues/6856.

Reply:
```suggestion
	// TODO: According to the spec of gateway API, the original scheme should be kept if no scheme is given.
	// So the logic may be changed if Kong gateway supports:
	// https://github.com/Kong/kubernetes-ingress-controller/issues/6856
```